### ISSUE TITLE
A new command: poetry test

### ISSUE THREAD

**Issue by ThatXliner** [python-poetry/poetry#3386]
- [X] I have searched the [issues](https://github.com/python-poetry/poetry/issues) of this repo and believe that this is not a duplicate.
- [X] I have searched the [documentation](https://python-poetry.org/docs/) and believe that my question is not covered.

## Feature Request

Implement a `poetry test` command.

### Background
I've been using Makefiles (and sometimes just snippets of code) to manage testing over CIs and I'd like a `poetry test` command to ~~manage~~ centralize this (for some systems don't have GNU Make).

### Demonstration
To configure it, there should be a new section in the` pyproject.toml` stating what `poetry test` does, like this:

```toml
[tool.poetry.test]
command = "pytest -vvv --cov=package_name"
after_success = 'echo "Yay! All tests have passed!" '
```

### Implementation

Basically, poetry will run that command in a virtual environment (which does not include the package itself) that has installed *all the dependancies*.

**Comment by ThatXliner** [comment]
While there is `tox`, `poetry test` would be a shorthand for `poetry install && poetry run <test command>`

**Comment by sinoroc** [comment]
If anything like that, then I'd rather see a general purpose shortcut for commands such as:

* https://pipenv.pypa.io/en/latest/advanced/#custom-script-shortcuts

**Comment by ThatXliner** [comment]
hmm, that would be nice!

**Comment by ThatXliner** [comment]
I might make a new issue proposing *that* instead. That's actually what I want: a centralized command shortcut (basically Makefiles except Makefiles are annoying)

**Comment by sinoroc** [comment]
There is always [_pyinvoke_](https://www.pyinvoke.org/)

I believe something like that could work (untested, pseudo code):

```
[tool.poetry.dev-dependencies]
invoke = '*'
```

```
import invoke

@invoke.task
def test(c):
    c.run('python -m pytest')
```

```
poetry run invoke test
```

---

Also maybe there are already tickets for a similar feature, I have not searched yet.

**Comment by sinoroc** [comment]
Related: https://github.com/python-poetry/poetry/pull/591#issuecomment-504762152

> Poetry is a package and dependency manager, not a task manager. This feature is beyond the scope and the original purpose of Poetry and will likely never be integrated into it.

I'd say (as I commented on many other feature requests), wait until _poetry_ gets its [plugin system](https://github.com/python-poetry/poetry/pull/1237) (scheduled for [_v1.2_](https://github.com/python-poetry/poetry/projects/6)), and this feature could hopefully be made into a poetry plugin.

**Comment by ThatXliner** [comment]
Got it, I'll use pyinvoke for now

**Comment by sinoroc** [comment]
Suggestion: [_poethepoet_](https://pypi.org/project/poethepoet/).
